src/external/visualization_integration.cu

```diff
@@ -29,7 +31,49 @@ __host__ std::string vector_double_to_json_array(const std::vector<double> &v)
     return ans;
 }
 
-__host__ bool send_particles_to_visualization(const std::string &url, MapNode *nodes, int n_of_nodes)
+__host__ bool send_model_to_visualization(const std::pair<std::string, std::string> &urls, Polyhedron *polyhedron)
+{
+    std::vector<double> polyhedron_vertices, polyhedron_faces;
+    std::vector<std::vector<double>> poly;
+
+    for(int i = 0; i < polyhedron->get_n_of_faces(); ++i)
+    {
+        for(int j = 0; j < polyhedron->get_faces()[i].get_n_of_vertices(); ++j)
+        {
+            SpacePoint v = polyhedron->get_faces()[i].get_vertices()[j];
+            poly.push_back({v.x, v.y, v.z});
+        }
+    }
+
+    std::string body = "[ ";
+    for(int i = 0; i < poly.size(); ++i)
+    {
+        body += vector_double_to_json_array(poly[i]);
+        if (i != poly.size() - 1)
+            body += ',';
+    }
+    body += " ]";
+
+    http::Request poly_request(urls.second);
+
+    try
+    {
+        const http::Response response = poly_request.send("POST", body, {"Content-Type: application/json"});
+
+        if(response.status < 200 || 300 <= response.status)
+            throw http::ResponseError("Response status is not OK");
+    }
+    catch(const std::exception &e)
+    {
+        std::cerr << "Request failed, error: " << e.what() << std::endl;
+        return false;
+    }
+
+    return true;
+}
+
+__host__ bool send_particles_to_visualization(const std::pair<std::string, std::string> &urls, MapNode *nodes,
```

Comment:
The second argument's type should be `const MapNode *` instead of the current one because the `nodes` array is not being modified inside of the function